### DISCUSSION TITLE
Is there a way to use multiple modals on a global state?

### DISCUSSION THREAD

**Discussion by alejandroGonGon** [heroui-inc/heroui#871] [Q&A] [answered]
Hi! I'm now using the NEXTUI framework, and I find myself interested in how the modals work, so the problem is that I'm not being able to have multiple modals and pass them props in a global state. 

Like I'm trying to access a modal from different pages so I can't have it in one, and at the same time, I need to pass it props also, anyone have tried this before with this UI library?

**Comment by Toddynn** [comment, marked as answer]
function MyComponent() {
  const modal1 = useDisclosure()
  const modal2 = useDisclosure()
  
  return (
    <>
      <Button onClick={modal1.onOpen}>Open Modal 1</Button>
      <Button onClick={modal2.onOpen}>Open Modal 2</Button>

      <Modal isOpen={modal1.isOpen} onClose={modal1.onClose}>
        // omitted for brevity
      </Modal>
      
      <Modal isOpen={modal2.isOpen} onClose={modal2.onClose}>
        // omitted for brevity
      </Modal>
    </>
  )
}

**Reply by hasnat5** [reply]
This works on my project, thanks. Very helpful
